app/src/main/java/org/simple/clinic/bp/assignbppassport/BpPassportSheet.kt

```diff
@@ -1,13 +1,128 @@
 package org.simple.clinic.bp.assignbppassport
 
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
 import android.os.Bundle
-import org.simple.clinic.R
+import com.jakewharton.rxbinding3.view.clicks
+import io.github.inflationx.viewpump.ViewPumpContextWrapper
+import io.reactivex.Observable
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.databinding.SheetBpPassportBinding
+import org.simple.clinic.di.InjectorProviderContextWrapper
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.util.wrap
 import org.simple.clinic.widgets.BottomSheetActivity
+import javax.inject.Inject
 
-class BpPassportSheet : BottomSheetActivity() {
+class BpPassportSheet : BottomSheetActivity(), BpPassportUiActions {
+
+  companion object {
+    private const val KEY_BP_PASSPORT_NUMBER = "bpPassportNumber"
+    private const val FACILITY_CHANGE = "alertFacilityChange"
+
+    fun selectedFacility(data: Intent): String? {
+      return data.getStringExtra(FACILITY_CHANGE)
+    }
+  }
+
+  @Inject
+  lateinit var effectHandlerFactory: BpPassportEffectHandler.Factory
+
+  private lateinit var component: BpPassportSheetComponent
+
+  private lateinit var binding: SheetBpPassportBinding
+
+  private val registerNewPatientButton
+    get() = binding.registerNewPatientButton
+
+  private val addToExistingPatientButton
+    get() = binding.addToExistingPatientButton
+
+  private val events: Observable<BpPassportEvent> by unsafeLazy {
+    Observable
+        .merge(
+            registerNewPatientClicks(),
+            addToExistingPatientClicks()
+        )
+  }
+
+  private val delegate by unsafeLazy {
+
+    MobiusDelegate.forActivity(
+        events = events,
+        defaultModel = BpPassportModel(),
```

Comment:
We cannot create a `BpPassportModel` like this, it has a constructor param here as well. So let's try to use the factory method here.

Reply:
This is updated in the prefilling commit. Please check

Reply:
Yes, but this commit will have a compilation error won't it? Because we are not providing a constructor param for the `BpPassportModel`